apps/web/src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/modules/DashboardManageVariableOptionsField.vue

```diff
@@ -0,0 +1,166 @@
+                    <p-text-input :value="option.key"
+                                  class="option-input"
+                                  :placeholder="$t('DASHBOARDS.CUSTOMIZE.VARIABLES.PLACEHOLDER_OPTIONS')"
+                                  :invalid="option.error"
+                                  @update:value="handleChangeOptionValue(index, $event)"
+                    />
+                    <span class="option-colon">:</span>
+                    <p-text-input :value="option.label"
+                                  class="option-input"
+                                  :placeholder="$t('Enter option label')"
+                                  :invalid="option.error"
+                                  @update:value="handleChangeOptionLabel(index, $event)"
+                    />
+                    <div class="option-delete-area">
+                        <p-icon-button v-if="options.length > 1"
+                                       name="ic_delete"
+                                       @click="handleDeleteOption(option.key)"
+                        />
+                    </div>
+                </div>
+            </draggable>
+        </div>
+    </p-field-group>
+</template>
+
+<script setup lang="ts">
+
+import { reactive } from 'vue';
+import draggable from 'vuedraggable';
+
+import {
+    PButton, PFieldGroup, PIconButton, PRadio, PRadioGroup, PTextInput, PI, useProxyValue,
+} from '@spaceone/design-system';
+
+import { getUUID } from '@/lib/component-util/getUUID';
+
+import type { OptionItem } from '@/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/type';
+
+
+interface Props {
+    options: OptionItem[];
+    isManualOptionsType: boolean;
+}
+interface EmitFn {
+    (e: string, value: string): void;
+    (e: 'save-click', value: string): void;
+    (e: 'cancel-click'): void;
+}
+const props = defineProps<Props>();
+const emit = defineEmits<EmitFn>();
+
+const state = reactive({
+    proxyOptions: useProxyValue<OptionItem[]>('options', props, emit),
+    isManualOption: true,
+});
+
+const handleChangeOptionValue = (index: number, value: string) => {
+    // TODO: reactor with planning
+    // state.proxyOptions[index].error = state.proxyOptions.some((option, _index) => {
+    //     if (index === _index || value === '') return false;
+    //     return option.key === value;
+    // });
+    state.proxyOptions[index].key = value;
+};
+const handleChangeOptionLabel = (index: number, value: string) => {
+    // TODO: reactor with planning
+    // state.proxyOptions[index].error = state.proxyOptions.some((option, _index) => {
+    //     if (index === _index || value === '') return false;
+    //     return option.label === value;
+    // });
+    state.proxyOptions[index].label = value;
+};
+const handleAddOption = () => {
+    state.proxyOptions = [...state.proxyOptions, { _key: getUUID(), key: '', label: '' }];
+};
+const handleDeleteOption = (_key: string) => {
```

Comment:
suggestion: I think it would be better to manage by index.

```suggestion
const handleDeleteOption = (index: number) => {
```

Reply:
Reflected just now!